ScanCommand/src/main/java/io/ballerina/scan/ScanCmd.java

```diff
@@ -114,8 +121,46 @@ public void execute() {
             return;
         }
 
+        Project project = getProject();
+        if (project == null) {
+            return;
+        }
+
+        ProjectAnalyzer projectAnalyzer = new ProjectAnalyzer();
+
         outputStream.println();
         outputStream.println("Running Scans...");
+
+        List<Issue> issues = projectAnalyzer.analyzeProject(project);
+
+        if (platforms.isEmpty() && !platformTriggered) {
+            ScanUtils.printToConsole(issues, outputStream);
+
+            if (project.kind().equals(ProjectKind.BUILD_PROJECT)) {
+                Path reportPath;
+                if (targetDir != null) {
+                    reportPath = ScanUtils.saveToDirectory(issues, project, targetDir);
+                } else {
+                    reportPath = ScanUtils.saveToDirectory(issues, project, null);
+                }
```

Comment:
Why can't we do this?

```suggestion
                reportPath = ScanUtils.saveToDirectory(issues, project, targetDir);
```